packages/grid/x-data-grid/src/components/menu/columnMenu/default/GridColumnMenuDefault.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import Divider from '@mui/material/Divider';
+import { useGridColumnMenuPreProcessing } from '../../../../hooks/features/columnMenu/useGridColumnMenuPreProcessing';
+import { GridColumnMenuDefaultContainer } from './GridColumnMenuDefaultContainer';
+import { GridColumnMenuProps } from '../GridColumnMenuProps';
+import { GridColumnMenuColumnsItem } from './GridColumnMenuColumnsItem';
+import { GridColumnMenuFilterItem } from './GridColumnMenuFilterItem';
+import { GridColumnMenuHideItem } from './GridColumnMenuHideItem';
+import { GridColumnMenuSortItem } from './GridColumnMenuSortItem';
+import { useGridPrivateApiContext } from '../../../../hooks/utils/useGridPrivateApiContext';
+
+export const COLUMN_MENU_DEFAULT_COMPONENTS = {
+  ColumnMenuSortItem: GridColumnMenuSortItem,
+  ColumnMenuFilterItem: GridColumnMenuFilterItem,
+  ColumnMenuHideItem: GridColumnMenuHideItem,
+  ColumnMenuColumnsItem: GridColumnMenuColumnsItem,
+};
+
+export const COLUMN_MENU_DEFAULT_COMPONENTS_PROPS = {
+  ColumnMenuSortItem: { displayOrder: 0 },
+  ColumnMenuFilterItem: { displayOrder: 10 },
+  ColumnMenuHideItem: { displayOrder: 20 },
+  ColumnMenuColumnsItem: { displayOrder: 30 },
+};
+
+const GridColumnMenuDefault = React.forwardRef<HTMLUListElement, GridColumnMenuProps>(
+  function GridColumnMenuDefault(props, ref) {
+    const {
+      defaultComponents = COLUMN_MENU_DEFAULT_COMPONENTS,
+      defaultComponentsProps = COLUMN_MENU_DEFAULT_COMPONENTS_PROPS,
```

Comment:
```suggestion
      defaultComponentsProps = COLUMN_MENU_DEFAULT_COMPONENTS_PROPS,
      components,
      componentsProps,
```

Those props are passed down to the DOM element and there's a console warning